lib/test.js

```diff
@@ -143,6 +218,20 @@ class Test {
 		this.pendingThrowsAssertion = null;
 		this.planCount = null;
 		this.startedAt = 0;
+		this.pendingAttemptCount = 0;
+		this.attemptCount = 0;
+
+		this.forAttempt = fn => {
+			const opts = Object.assign({}, options, {
+				metadata: Object.assign({}, options.metadata, {
+					callback: false,
+					failing: false
+				}),
+				fn,
+				title: this.title + '.A' + (this.attemptCount++)
+			});
+			return new Test(opts);
```

Comment:
I was adjusting the title in order to prevent any conflicts related to snapshots. Isn't it the case that if we adjust the title, then there are no problems about number of concurrent `t.try()` and both may contain snapshots?

Reply:
The snapshots are indexed on the title, so if currently you only commit the third attempt, then later if the first attempt is committed you won't actually compare the snapshot. See also #1585.